### PR TITLE
SALTO-1778-workflow-bug

### PR DESCRIPTION
**Pull request by alonstern**
Fixed two issues in workflow deployment
- When creating a workflow it automatically adds an additional `PermissionValidator` with `permissionKey` of `CREATE_ISSUES`  to the initial transition so if it's already in the NaCl I remove it to not create duplicate validators

- Jira does not support deploying script runners config so I removed it from the nacl on deploy and added a changed validator

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/workflow/workflow.ts [hunk 1]
```diff
@@ -13,19 +13,32 @@
...
+  'com.onresolve.jira.groovy.groovyrunner__script-workflow-validators',
+]
+
+export const UNDEPLOYALBE_POST_FUNCTION_TYPES = [
+  'com.onresolve.jira.groovy.groovyrunner__script-postfunction',
+]
+
 // Taken from https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-workflows/#api-rest-api-3-workflow-post
-const SUPPORTED_POST_FUNCTIONS_TYPES = [
+const FETCHED_POST_FUNCTIONS_TYPES = [
```
Comment (netama) [inline packages/jira-adapter/src/filters/workflow/workflow.ts:41]: tiny nit: this has `functions` and below there is `function`, I assume `function` is more accurate?

File: packages/jira-adapter/src/filters/workflow/workflow.ts [hunk 2]
```diff
@@ -121,8 +136,73 @@ const transformWorkflowInstance = (workflowValues: Workflow): void => {
     .forEach(transition => transformRules(transition.rules as Rules, transition.type))
 }
 
+/**
+ * When creating a workflow, the initial transition is always created
+ * with an extra PermissionValidator with CREATE_ISSUES permission key.
+ * Currently the API does not allow us to remove it but we can at least make sure to
+ * not create an additional one if one validator like that is already appears in the nacl.
```
Comment (netama) [inline packages/jira-adapter/src/filters/workflow/workflow.ts:143]: nit: remove `is`

File: packages/jira-adapter/src/filters/workflow/workflow.ts [hunk 3]
```diff
@@ -121,8 +136,73 @@ const transformWorkflowInstance = (workflowValues: Workflow): void => {
...
+          {
+            type: 'PermissionValidator',
+            configuration: {
+              permissionKey: 'CREATE_ISSUES',
+            },
+          },
+        ),
+      )
+
+      _.pullAt(
```
Comment (netama) [inline packages/jira-adapter/src/filters/workflow/workflow.ts:162]: nit - discussed IRL - might be better to use one of the more common functions

File: packages/jira-adapter/test/filters/workflow_values.ts [hunk 4]
```diff
@@ -153,3 +153,73 @@ export const WITH_VALIDATORS = {
     },
```
Comment (netama) [inline packages/jira-adapter/test/filters/workflow_values.ts:153]: nit: missing `.test` in the filename

**Review by netama — APPROVED** [review]

File: packages/jira-adapter/test/filters/workflow_values.ts (hunk 4 above)
Comment (alonstern) [reply packages/jira-adapter/test/filters/workflow_values.ts:153]: This has no tests in it. Do we usually add `.test` for this files as well?

**Pull request merged**